docs/developer/automerge.md

```diff
@@ -0,0 +1,25 @@
+# Automatic Merging of Pull Requests in OSM
+This document aims to explain how OSM utilizes the [automerge-action bot](https://github.com/pascalgn/automerge-action) to automatically merge pull requests and how it was set up.
+
+## Workaround to triggering workflows
+By default the automerge-action bot uses a `GITHUB_TOKEN` to perform automerging; however, events triggered via `GITHUB_TOKEN` will not further trigger additional workflow runs. To work around this, we created a GitHub App (Generate_Token) to generate a token that is then used to create a JSON Web Token (JWT) that authenticates and retrieves the Installation Access Token (IAT) from the GitHub API. Passing in the IAT as the `GITHUB_TOKEN`, will allow both `on:push` and `on:pull_request` workflows to trigger.
+
+## How it works
+- Automerge only occurs when the `automerge` label is assigned. To prevent automerging, you can either not assign the `automerge` label or include one of the following labels: `wip` or `do-not-merge/hold`.
```

Comment:
Are there any rules of thumb we should follow for when and when not to use the `automerge` label?

Reply:
It is up to the maintainer at the moment to decide whether a specific PR can be automerged. I think a few rules should be documented regarding when the automerge label can be used. We haven't discussed this yet.

There are several cases where an automerge label will do more harm than good, especially when a PR needs more than just the minimum 2 reviews. Since it is not standard practice to put the `do-not-merge/hold` label for every PR that is pending reviews by default, we need to think through and outline in this doc on when a PR can safely use the automerge label.